src/search/instamove/instamove.cc

```diff
@@ -140,23 +142,32 @@ class ValueHeadSearch : public InstamoveSearch {
 
     const size_t best_idx =
         std::min_element(results.begin(), results.end(),
-                         [](const EvalResult& a, const EvalResult& b) {
-                           return a.q < b.q;
+                         [](const auto& a, const auto& b) {
+                           if (a.first.has_value() != b.first.has_value()) {
+                             return a.first.has_value();
+                           }
+                           if (a.first != b.first) {
+                             return a.first.value() < b.first.value();
+                           }
+                           return a.second.q < b.second.q;
```

Comment:
I can't figure out how this covers every combination, but didn't manage to trip it up either. Either needs a comment or I need more coffee...